File: server/python/hosted_tokenization/hosted_tokenization_controller.py
from flask import request # type: ignore
from hosted_tokenization import hosted_tokenization_mapper, hosted_tokenization_service
from payment import create_payment_service, payment_details_service

def initialize_hosted_tokenization():
    hosted_tokenization_request = hosted_tokenization_service.init_hosted_tokenization()
    return hosted_tokenization_request.to_dictionary()

def create_hosted_tokenization(hosted_tokenization_request_dto):
    hosted_tokenization_request = hosted_tokenization_mapper.to_hosted_tokenization_request(hosted_tokenization_request_dto)
    create_payment_response = create_payment_service.create_payment(hosted_tokenization_request)
    return create_payment_response.to_dictionary()

def outcome():
    paymentId = request.args.get('paymentId')
    payment_details_response = payment_details_service.get_payment_details(paymentId)
    return payment_details_response.to_dictionary()

File: server/ruby/app/controllers/hosted_tokenization_controller.rb
require_relative '../hosted_tokenization/hosted_tokenization_mapper'
require_relative '../hosted_tokenization/hosted_tokenization_service'
require_relative '../payment/create_payment_service'
require_relative '../payment/payment_details_service'
require_relative '../configuration/merchant_client_config'
require_relative '../services/payment_utils'
require 'onlinepayments/sdk'


class HostedTokenizationController < ApplicationController

  def initialize_hosted_tokenization
    dto = HostedTokenizationService.init_hosted_tokenization()
    render json: PaymentUtils.to_camel_case_keys(dto.to_h)
  end

  def create_hosted_tokenization
    create_hosted_tokenization_basic_dto = CreateHostedTokenizationBasicDto.new(
      input_params[:amount],
      input_params[:currency],
      input_params[:hostedTokenizationId]
    )

    create_payment_request = HostedTokenizationMapper.to_hosted_tokenization_payment_request(create_hosted_tokenization_basic_dto)
    create_payment_response = CreatePaymentService.create_payment(create_payment_request)
    render json: create_payment_response
  end

  def outcome
    payment_id = params[:paymentId]

    payment_details_response = PaymentDetailsService.get_payment_details(payment_id)
    render json: payment_details_response
  end

  private

  def input_params
    params.permit(:amount, :currency, :redirectUrl)
  end

end
